src/Components/ServiceScene/Breakdowns/PatternsBreakdownScene.tsx

```diff
@@ -188,19 +186,121 @@ export class PatternsBreakdownScene extends SceneObjectBase<PatternsBreakdownSce
             isLazy: true,
             children: children.map((child) => child.clone()),
           }),
+          this.getSingleViewLayout(patternFrames, logExploration),
         ],
       }),
       loading: false,
     });
   }
 
-  private buildPatterns(patterns: LokiPattern[], appliedPatterns?: AppliedPattern[]) {
+  private extendTimeSeriesLegendBus(vizPanel: VizPanel, context: PanelContext) {
+    const originalFn = context.onToggleSeriesVisibility;
+
+    context.onToggleSeriesVisibility = (label: string, mode: SeriesVisibilityChangeMode) => {
+      originalFn?.(label, mode);
+
+      const override: ConfigOverrideRule | undefined = vizPanel.state.fieldConfig.overrides?.[0];
+      const patternsToShow: string[] = override?.matcher.options.names;
+      const legendSyncPatterns = new Set<string>();
+
+      if (patternsToShow) {
+        patternsToShow.forEach(legendSyncPatterns.add, legendSyncPatterns);
+      }
+
+      this.setState({
+        legendSyncPatterns,
+      });
+    };
+  }
+
+  private getSingleViewLayout(patternFrames: PatternFrame[], logExploration: IndexScene) {
+    const appliedPatterns = sceneGraph.getAncestor(logExploration, IndexScene).state.patterns;
+    const timeRange = sceneGraph.getTimeRange(this).state.value;
+    const timeSeries = PanelBuilders.timeseries()
+      .setData(
+        new SceneDataNode({
+          data: {
+            series: patternFrames.map((patternFrame) => {
+              return patternFrame.dataFrame;
+            }),
+            state: LoadingState.Done,
+            timeRange: timeRange,
+          },
+        })
+      )
+      .setOption('legend', {
+        asTable: true,
+        showLegend: true,
+        displayMode: LegendDisplayMode.Table,
+        placement: 'right',
+        width: 200,
+      })
+
+      .setTitle('Patterns')
+      .setLinks([
+        {
+          url: '',
+          onClick: (event) => {
+            onPatternClick({
+              pattern: event.origin.name,
+              type: 'include',
+              indexScene: logExploration,
+            });
+          },
+          title: 'Select',
+        },
+        {
+          url: '',
+          onClick: (event) => {
+            onPatternClick({
+              pattern: event.origin.name,
+              type: 'exclude',
+              indexScene: logExploration,
+            });
+          },
+          title: 'Exclude',
+        },
+      ])
+      .build();
+
+    timeSeries.setState({
+      extendPanelContext: (vizPanel, context) => this.extendTimeSeriesLegendBus(vizPanel, context),
+    });
+
+    return new SceneFlexLayout({
+      direction: 'column',
+      width: 'calc(100vw - 60px)',
+      maxWidth: '100%',
+      children: [
+        new SceneFlexLayout({
```

Comment:
Is this `SceneFlexLayout` within a `SceneFlexLayout` necessary? Can't it be another `SceneFlexItem` inside the first one?